### DISCUSSION TITLE
Would Date.toLocaleString() return the servers date or the clients date?

### DISCUSSION THREAD

**Discussion by rigba** [vercel/next.js#27098] [Help]
Im not too sure how to even go about testing this but I was thinking since Nextjs is fetched on a server if I were to have some code like this for example.

```ts
import React, { FC } from 'react'

interface dsadaProps {

}

const dsada: FC<dsadaProps> = ({}) => {
  return (
    <>
      <div>{Date.toLocaleString())}</div>
    </>
  );
}

export default dsada
```

would it grab the servers or clients date. If it did grab the servers date would I need to grab the date via `getStaticProps ` instead?

**Comment by SidKH** [comment]
### Rendering in JSX
If you try to render a date in JSX, like in your code snippet, Next will show you a hydration warning:
`Warning: Text content did not match. Server: "7/12/2021, 9:55:56 PM" Client: "7/13/2021, 12:55:58 AM"`
And will end up rendering the client time.

### Using Next.js functions
If you try to put it inside `getStaticProps`, it will give you a server date **at the build time**, which is probably not what you want.  
The `getServerSideProps` will return a server date at **request time**.

### Client-side
If you want to use client-side date without any warnings you can use `useEffect`
```jsx
const [date, setDate] = useState('')
  
useEffect(() => {
  const currentDate = new Date()
  setDate(currentDate. toLocaleString())
}, [])
  
return <div>{date}</div>
```

### Server side date without SSR
Let's say you want a server date but don't want to enable SSR with `getServerSideProps`.  
In this case, you can create Next.js API route that will return date from the server. Then fetch the date from this API route on the client in `useEffect`.
